src/templates/component/TemplateComponent/TemplateComponent.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2012-2022  Online-Go.com
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import * as React from "react";
+
+import * as data from "data";
+
+import { pgettext } from "translate";
+
+type TemplateComponentProps = {
+    render_twice?: boolean;
+};
+
+export function TemplateComponent(props: TemplateComponentProps): JSX.Element {
+    const [username, setUsername] = React.useState("");
+
+    const hangs_around = React.useRef("forever");
+
+    const user = data.get("config.user"); // user.anonymous will be set if they aren't logged in
```

Comment:
better to use `const user = useUser();` these days, comes from `import { useUser } from "hooks";`

Reply:
😮    ... and another reason to have learnings in a template!

Reply:
(useUser will automatically refresh when the user object is updated, as in when the username/icon/ui_class/rank changes kinda stuff)

Reply:
Yeah it's pretty new, figured if we're making a template though, that should be in there :)

Reply:
Template's been there <24 hours, already winning :D  :D